doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,182 @@
+)
+from skimage.data import palisades_of_vogt
+
+
+#####################################################################
+# The dataset we are using in this example is an image sequence (a movie!) of
+# human in-vivo tissue. Specifically, it shows the *palisades of Vogt* of a
+# given cornea sample.
+
+#####################################################################
+# Load image data
+# ===============
+
+image_seq = palisades_of_vogt()
+
+print(f'number of dimensions: {image_seq.ndim}')
+print(f'shape: {image_seq.shape}')
+print(f'dtype: {image_seq.dtype}')
+
+#####################################################################
+# The dataset is a timeseries of 60 frames (which are 2D images). We can
+# visualize it by taking advantage of the ``animation_frame`` parameter in
+# Plotly's ``imshow`` function.
+
+fig = px.imshow(
+    image_seq,
+    animation_frame=0,
+    height=500,
+    width=500,
+    binary_string=True,
+    labels={'animation_frame': 'time point'},
+    title='Sample of in-vivo human cornea'
+)
+plotly.io.show(fig)
+
+#####################################################################
+# Average over time
+# =================
+# First, we want to detect those dark spots where the data are lost. In
+# technical terms, we want to *segment* the dark spots. We want to do so for
+# all frames in the sequence. Unlike the actual data (signal), the dark spots
+# do not move from one frame to the next; they are still. Therefore, we begin
+# by computing the time average of the image sequence. We shall use this
+# time-averaged image to segment the dark spots, the latter then standing out
+# with respect to the background (blurred signal).
+
+image_avg = np.mean(image_seq, axis=0)
+
+print(f'shape: {image_avg.shape}')
+
+fig = px.imshow(
+    image_avg,
+    width=500,
+    height=500,
+    binary_string=True,
+    title='Time-averaged image'
+)
+plotly.io.show(fig)
+
+#####################################################################
+# Use local thresholding
+# ======================
+# To segment the dark spots, we use thresholding. The images we are working
+# with have uneven illumination, which causes variations in the (absolute)
+# intensities of the foreground and the background, from one region to another
+# (distant) one. It is therefore more fitting to compute different threshold
+# values across the image, one for each region. This is called adaptive (or
+# local) thresholding, as opposed to the usual thresholding procedure which
+# employs a single (global) threshold for all pixels in the image.
+#
+# Let us compare the visibility of the hidden data in two different
+# thresholding masks, one of which has the `block_size` set to 21, while
+# the other has it set to 35. For this, we start by defining a convenience
+# function to create a mask:
+
+def create_mask(image_avg, spot_size):
+    thresh_value = filters.threshold_local(
+        image_avg,
+        block_size=spot_size,
+        offset=10
```

Comment:
We probably want to add a comment explaining how we chose this value for the `offset` parameter.